mhs/common/mhs_common/workflow/common.py

```diff
@@ -71,6 +74,24 @@ def __init__(self, routing: routing_reliability.RoutingAndReliability = None):
                            {'Exception': e})
             raise e
 
+    async def _lookup_to_asid_details(self, interaction_details: Dict):
```

Comment:
This seems like unnecessary duplication. Couldn't you just add the call to `_extract_asid` into `_lookup_endpoint_details`? It would probably be worth refactoring it slightly to return an object/dictionary, rather than extending the tuple returned.